### PR TITLE
Add a help pages detailing out OpenXR settings.

### PR DESCRIPTION
**Pull request by BastiaanOlij**
This is a new help page for the XR part of the manual that details out all the settings available for OpenXr.

(this started out life as just detailing out the reference spaces... )

### CODE REVIEW

File: tutorials/xr/openxr_settings.rst [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+------------------
+
+This specifies the path of the action map file that OpenXR will load and communicate to the XR Runtime.
+
+Form Factor
+-----------
+
+This specifies whether your game is designed for:
+
+- ``Head Mounted`` devices such as a Meta Quest, Valve Index or Magic Leap,
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:33]:
```suggestion
- ``Head Mounted`` devices such as a Meta Quest, Valve Index, or Magic Leap,
```

File: tutorials/xr/openxr_settings.rst [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+This specifies the view configuration your game is designed for:
+
+- ``Mono``, your game provides a single image output. E.g. phone based AR;
+- ``Stereo``, your game provides stereo image output. E.g. head mounted devices.
+
+If the device on which you run your game does not match the selection here, OpenXR will fail to initialise.
+
+.. note::
+  OpenXR has additional view configuration for very specific devices that Godot currently doesn't support yet.
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:49]:
```suggestion
  OpenXR has additional view configuration for very specific devices that Godot doesn't support yet.
```
Or
```suggestion
  OpenXR has additional view configuration for very specific devices that Godot currently doesn't support.
```

File: tutorials/xr/openxr_settings.rst [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+This reference space however does not prevent the user from walking away so you will need to detect if the user does so
+if you wish to prevent the user from leaving the vehicle they are controlling, which could potentially be game breaking.
+
+This reference space is the best option for games like flight simulators or racing simulators
+where we want to place the :ref:`XROrigin3D <class_xrorigin3d>` node where the players head should be.
+
+When the user enacts the recenter option on their headset, the method of which is different per XR runtime,
+the XR runtime will move the :ref:`XRCamera3D <class_xrcamera3d>` to the :ref:`XROrigin3D <class_xrorigin3d>` node.
+The :ref:`OpenXRInterface <class_openxrinterface>` will also omit the ``pose_recentered`` signal
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:76]:
```suggestion
The :ref:`OpenXRInterface <class_openxrinterface>` will also emit the ``pose_recentered`` signal
```

File: tutorials/xr/openxr_settings.rst [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+Other XR runtimes may deside on the placement of this point by other means.
+It is however a stationary point in the real world.
+
+This reference space is the best option for room scale games where the user is expected to walk around a larger space,
+or for games where there is a need to switch between game modes.
+See :ref:`Room Scale <doc_xr_room_scale>` for more information.
+
+When the user enacts the recenter option on their headset, the method of which is different per XR runtime,
+the XR runtime will not change the origin point.
+The :ref:`OpenXRInterface <class_openxrinterface>` will omit the ``pose_recentered`` signal
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:99]:
```suggestion
The :ref:`OpenXRInterface <class_openxrinterface>` will emit the ``pose_recentered`` signal
```

File: tutorials/xr/openxr_settings.rst [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+and it is up to the game to react appropriately.
+Not doing so will prevent your game from being accepted on various stores.
+
+In Godot you can do this by calling the ``center_on_hmd`` function on the  :ref:`XRServer <class_xrserver>`:
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:103]:
```suggestion
In Godot you can do this by calling the ``center_on_hmd`` function on the :ref:`XRServer <class_xrserver>`:
```

File: tutorials/xr/openxr_settings.rst [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+The player is thus also free to walk away.
+
+This reference space is the best option of games where the user is expected to stand in the same location
+or for AR type games where the users interface elements are bound to the origin node
+and are quickly placed at the players location on recenter.
+
+When the user enacts the recenter option on their headset, the method of which is different per XR runtime,
+the XR runtime will move the :ref:`XRCamera3D <class_xrcamera3d>` above the :ref:`XROrigin3D <class_xrorigin3d>` node
+but keeping the players height.
+The :ref:`OpenXRInterface <class_openxrinterface>` will also omit the ``pose_recentered`` signal
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:131]:
```suggestion
The :ref:`OpenXRInterface <class_openxrinterface>` will also emit the ``pose_recentered`` signal
```

File: tutorials/xr/openxr_settings.rst [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+In this mode however the height of the player is kept.
+Same as with the local reference space, some XR runtimes will persist this location over sessions.
+
+It is thus not guaranteed the player will be standing on the origin point,
+the only guarantee is that they were standing there when the user last recentered.
+The player is thus also free to walk away.
+
+This reference space is the best option of games where the user is expected to stand in the same location
+or for AR type games where the users interface elements are bound to the origin node
+and are quickly placed at the players location on recenter.
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:126]:
```suggestion
and are quickly placed at the player's location on recenter.
```

File: tutorials/xr/openxr_settings.rst [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+but keeping the players height.
+The :ref:`OpenXRInterface <class_openxrinterface>` will also omit the ``pose_recentered`` signal
+so your game can react accordingly.
+
+.. Warning::
+  Be careful using this mode in combination with virtual movement of the player.
+  The user recentering in this scenario can be unpredicable unless you counter the move when handling the recenter signal.
+  This can even be game breaking as the effect in this scenario would be the player teleporting to whatever abstract location
+  the origin point was placed at during virtual movement, including the ability for players teleporting into
+  location that should be off limits.
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:139]:
```suggestion
  locations that should be off limits.
```

File: tutorials/xr/openxr_settings.rst [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+Environment Blend Mode
+----------------------
+
+The environment blend mode defines how our rendered output is blended into "the real world" provided this is supported by the headset.
+
+- ``Opaque`` means our output obscures the real world, we are in VR mode.
+- ``Additive`` means our output is added to the real world,
+  this is an AR mode where optics do not allow us to fully obscure the real world (e.g. Hololens),
+- ``Alpha`` means our output is blended with the real world using the alpha output (viewport should have transparent background enabled),
+  this is an AR mode where optics can fully obscure the real world (Magic Leap, all pass through devices etc.).
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:157]:
```suggestion
  this is an AR mode where optics can fully obscure the real world (Magic Leap, all pass through devices, etc.).
```

File: tutorials/xr/openxr_settings.rst [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+In order to use this functionality you need to edit your action map and add a new pose action,
+say ``eye_pose``.
+Now add a new interaction profile for the eye gaze interaction and map the ``eye_pose``:
+
+.. image:: img/openxr_eye_gaze_interaction.webp
+
+Don't forget to save!
+
+Next add a new :ref:`XRController3D <class_xrcontroller3d>` node to your origin node
+and set it's ``tracker`` property to ``/user/eyes_ext``
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:250]:
```suggestion
and set its ``tracker`` property to ``/user/eyes_ext``
```

File: tutorials/xr/openxr_settings.rst [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+and set it's ``pose`` property to ``eye_pose``.
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:251]:
```suggestion
and set its ``pose`` property to ``eye_pose``.
```

File: tutorials/xr/openxr_settings.rst [hunk 12]
```diff
@@ -0,0 +1,253 @@
...
+The :ref:`OpenXRInterface <class_openxrinterface>` will omit the ``pose_recentered`` signal
+and it is up to the game to react appropriately.
+Not doing so will prevent your game from being accepted on various stores.
+
+In Godot you can do this by calling the ``center_on_hmd`` function on the  :ref:`XRServer <class_xrserver>`:
+
+- Calling ``XRServer.center_on_hmd(XRServer.RESET_BUT_KEEP_TILT, true)`` will move the :ref:`XRCamera3D <class_xrcamera3d>` node
+  to the :ref:`XROrigin3D <class_xrorigin3d>` node similar to the ``Local`` reference space.
+- Calling ``XRServer.center_on_hmd(XRServer.RESET_BUT_KEEP_TILT, true)`` will move the :ref:`XRCamera3D <class_xrcamera3d>` node
+  above the :ref:`XROrigin3D <class_xrorigin3d>` node keeping the players height, similar to the ``Local Floor`` reference space.
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:108]:
```suggestion
  above the :ref:`XROrigin3D <class_xrorigin3d>` node keeping the player's height, similar to the ``Local Floor`` reference space.
```

File: tutorials/xr/openxr_settings.rst [hunk 13]
```diff
@@ -0,0 +1,253 @@
...
+^^^^^
+
+The local reference space places our origin point at the players head by default.
+Some XR runtimes will do this each time your game starts, others will make the position persist over sessions.
+
+This reference space however does not prevent the user from walking away so you will need to detect if the user does so
+if you wish to prevent the user from leaving the vehicle they are controlling, which could potentially be game breaking.
+
+This reference space is the best option for games like flight simulators or racing simulators
+where we want to place the :ref:`XROrigin3D <class_xrorigin3d>` node where the players head should be.
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:72]:
```suggestion
where we want to place the :ref:`XROrigin3D <class_xrorigin3d>` node where the player's head should be.
```

File: tutorials/xr/openxr_settings.rst [hunk 14]
```diff
@@ -0,0 +1,253 @@
...
+Within XR all elements like the players head and hands are tracked within a tracking volume.
+At the base of this tracking volume is our origin point, which maps our virtual space to the real space.
+There are however different scenarios that place this point in different locations,
+this depends on the XR system used.
+In OpenXR these scenarios are well defined and selected by setting a reference space.
+
+Local
+^^^^^
+
+The local reference space places our origin point at the players head by default.
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:65]:
```suggestion
The local reference space places our origin point at the player's head by default.
```

File: tutorials/xr/openxr_settings.rst [hunk 15]
```diff
@@ -0,0 +1,253 @@
...
+
+.. note::
+  OpenXR has additional view configuration for very specific devices that Godot currently doesn't support yet.
+  For instance, Varjo headsets have a quad view configuration that outputs two sets of stereo images.
+  These may be supported in the near future. 
+
+Reference Space
+---------------
+
+Within XR all elements like the players head and hands are tracked within a tracking volume.
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:56]:
```suggestion
Within XR all elements like the player's head and hands are tracked within a tracking volume.
```

File: tutorials/xr/openxr_settings.rst [hunk 16]
```diff
@@ -0,0 +1,253 @@
...
+
+If your game only supports controllers this should be turned off. 
+
+See the chapter on :ref:`hand tracking <doc_openxr_hand_tracking>` for additional details.
+
+Eye Gaze Interaction
+^^^^^^^^^^^^^^^^^^^^
+
+This enables the eye gaze interaction extension when supported by the device used.
+When enabled we will get feedback from eye tracking through a pose situated between the users eyes
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:238]:
```suggestion
When enabled we will get feedback from eye tracking through a pose situated between the user's eyes
```

File: tutorials/xr/openxr_settings.rst [hunk 17]
```diff
@@ -0,0 +1,253 @@
...
+Extensions
+----------
+
+This subsection provides access to various optional OpenXR extensions.
+
+Hand Tracking
+^^^^^^^^^^^^^
+
+This enables the hand tracking extension when supported by the device used. This is on by default for legacy reasons.
+The hand tracking extension provides access to data that allows you to visualise the users hands with correct finger positions.
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:226]:
```suggestion
The hand tracking extension provides access to data that allows you to visualise the user's hands with correct finger positions.
```

File: tutorials/xr/openxr_settings.rst [hunk 18]
```diff
@@ -0,0 +1,253 @@
...
+the only guarantee is that they were standing there when the user last recentered.
+The player is thus also free to walk away.
+
+This reference space is the best option of games where the user is expected to stand in the same location
+or for AR type games where the users interface elements are bound to the origin node
+and are quickly placed at the players location on recenter.
+
+When the user enacts the recenter option on their headset, the method of which is different per XR runtime,
+the XR runtime will move the :ref:`XRCamera3D <class_xrcamera3d>` above the :ref:`XROrigin3D <class_xrorigin3d>` node
+but keeping the players height.
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:130]:
```suggestion
but keeping the player's height.
```

File: tutorials/xr/openxr_settings.rst [hunk 19]
```diff
@@ -0,0 +1,253 @@
...
+The local floor reference space is similar to the local reference space as it positions the origin point where the player is.
+In this mode however the height of the player is kept.
+Same as with the local reference space, some XR runtimes will persist this location over sessions.
+
+It is thus not guaranteed the player will be standing on the origin point,
+the only guarantee is that they were standing there when the user last recentered.
+The player is thus also free to walk away.
+
+This reference space is the best option of games where the user is expected to stand in the same location
+or for AR type games where the users interface elements are bound to the origin node
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:125]:
```suggestion
or for AR type games where the user's interface elements are bound to the origin node
```

File: tutorials/xr/openxr_settings.rst [hunk 20]
```diff
@@ -0,0 +1,253 @@
...
+Most XR runtimes only support fixed foveation, but some will take eye tracking into account and use the focal point for this effect.
+
+The higher the level, the better the performance gains, but also the more reduction in quality there is in the users peripheral vision.
+
+.. Note::
+  **Compatibility renderer only**,
+  for Mobile and Forward+ renderer, set the ``vrs_mode`` property on :ref:`Viewport <class_viewport>` to ``VRS_XR``.
+
+.. Warning::
+  This feature is disabled if post effects are used such as glow, bloom of DOF.
```
Comment (AThousandShips) [inline tutorials/xr/openxr_settings.rst:180]:
```suggestion
  This feature is disabled if post effects are used such as glow, bloom, or DOF.
```

File: tutorials/xr/openxr_settings.rst [hunk 21]
```diff
@@ -0,0 +1,253 @@
+.. _doc_openxr_settings:
+
+OpenXR Settings
+===============
+
+OpenXR has its own set of settings that are applied when OpenXR starts.
+It is possible for OpenXR extensions implemented through Godot plugins to add additional settings,
+we will only discuss the settings in the core of Godot here.
```
Comment (dsnopek) [inline tutorials/xr/openxr_settings.rst:8]:
```suggestion
While it is possible for OpenXR extensions implemented through Godot plugins to add additional settings,
we will only discuss the settings in the core of Godot here.
```

File: tutorials/xr/openxr_settings.rst [hunk 22]
```diff
@@ -0,0 +1,253 @@
...
+
+.. image:: img/openxr_settings.png
+
+Enabled
+-------
+
+This setting enables the OpenXR module when Godot starts.
+This is required when the Vulkan back end is used.
+For other back ends you can enable OpenXR at any time by calling ``initialize`` on the :ref:`OpenXRInterface <class_openxrinterface>`.
```
Comment (dsnopek) [inline tutorials/xr/openxr_settings.rst:17]:
```suggestion
For other backends, you can enable OpenXR at any time by calling ``initialize`` on the :ref:`OpenXRInterface <class_openxrinterface>`.
```

File: tutorials/xr/openxr_settings.rst [hunk 23]
```diff
@@ -0,0 +1,253 @@
...
+  For instance, Varjo headsets have a quad view configuration that outputs two sets of stereo images.
+  These may be supported in the near future. 
+
+Reference Space
+---------------
+
+Within XR all elements like the players head and hands are tracked within a tracking volume.
+At the base of this tracking volume is our origin point, which maps our virtual space to the real space.
+There are however different scenarios that place this point in different locations,
+this depends on the XR system used.
```
Comment (dsnopek) [inline tutorials/xr/openxr_settings.rst:59]:
```suggestion
There are however different scenarios that place this point in different locations,
depending on the XR system used.
```

**Comment by BastiaanOlij** [comment]
Thanks @AThousandShips , @dsnopek  :)

**Review by skyace65 — APPROVED** [review]

**Review by mhilbrunner — APPROVED** [review]

**Review by AThousandShips — APPROVED** [review]

**Comment by skyace65** [comment]
Thanks!

**Pull request merged**